### ISSUE TITLE
Crash in format_for_country() for invalid UK number

### ISSUE THREAD

**Issue by mitchjacksontech** [DrHyde/perl-modules-Number-Phone#98]
Am experiencing a crash in format_for_country() for a UK number.
The number was user-entered, and I think is invalid as a UK number.
A similar crashing number: `+445025259012`

Am I correct assuming format_for_country() should never crash?

This crash does not occur with known valid UK phone numbers, afaik.
I am using 3.6006. Problem also observed with 3.4003, where I think format_for_country() was introduced

To reproduce:

```perl
use Number::Phone;
use feature 'say';

# Prints 3.6006
say $Number::Phone::VERSION;

my $p = Number::Phone->new('+445025259012');

# prints 1
say $p->is_valid;

# prints +44 5025259012
say $p->format;

# Crash with error:
# Can't call method "country_code" on unblessed reference at .../Number/Phone/Formatter.pm line 20.
say $p->format_for_country('US');

```

Or as a test unit if preferred

```perl
use strict;
use Number::Phone;
use Test::More;
use Try::Tiny;

my @test_numbers = (

  # Test numbers from Number::Phone t/formaters.t
  qw/
    +442087712924
    +12024181440
    +81335803311
  /,

  # Possibly invalid UK number
  '+445025259012',
);



test_number($_) for @test_numbers;

done_testing();

sub test_number {
  my $n = shift;

  my ( $is_valid, $format, $format_for_country );
  try {
    my $p = Number::Phone->new( $n );
    $is_valid = $p->is_valid;
    $format = $p->format;
    $format_for_country = $p->format_for_country('US');
  };

  ok(
    $is_valid && $format && $format_for_country,
    sprintf '%s is_valid:%s format:%s format_for_country(US):%s',
      $n,
      $is_valid || 'ERROR',
      $format || 'ERROR',
      $format_for_country || 'ERROR',
  );
}

```

**Comment by DrHyde** [comment]
Thank you, this sort of thorough bug report is exactly what I like to see!

**Comment by DrHyde** [comment]
The number is indeed invalid, it is being incorrectly identified as valid but not allocated. You shouldn't even be able to construct invalid numbers like that! The fix is in 4f9c297, and will be in the next quarterly release, expected around the first week of December.
